models/logs.js

```diff
@@ -52,7 +54,34 @@ const fetchLogs = async (query, param) => {
   }
 };
 
+/**
+ * Fetches member cache Self logs
+ *
+ * @param userId { String }: Unique ID of the User
+ */
+const fetchMemberCacheLogs = async (id) => {
+  try {
+    const call = logsModel
+      .where("type", "==", logType.CLOUDFLARE_CACHE_PURGED)
+      .where("timestamp", ">=", getLast24HourTime(admin.firestore.Timestamp.fromDate(new Date())))
+      .where("meta.userId", "==", id);
+
+    const snapshot = await call.get();
+    const logs = [];
+    snapshot.forEach((doc) => {
```

Comment:
Is snapshot of type array? Even if there is no/problem with promise?

Reply:
It's a type of object, in which we get all the queried documents

Reply:
Your answer resolves second statement